### ISSUE TITLE
Poor docs

### ISSUE THREAD

**Issue by charleschege** [Luminarys/synapse#200]
This project is poorly documented, I have to go deep into the code to understand how everything works. Create better developer documentation

**Comment by evanrichter** [comment]
would you like to make suggestions for better documentation in the form of a pull request? @Luminarys would probably be open to that!

**Comment by charleschege** [comment]
I would like to help with the docs, but going through the code sometimes I get confused as to what some function arguments do! The easiest would be for you as developer to add doc comments to the structs and functions in the synapse library and sycli examples.

This would help a lot in understanding the code and could help others build tutorials or examples based off developer documentation

**Comment by Luminarys** [comment]
I fully agree, there's a lot of things which are poorly documented and have confusing behavior. In an ideal world I could spend a bunch of time rewriting things and properly documenting/testing them but sadly my time is limited.

That being said, I'm gradually going through parts of the code and trying to improve them. Currently I'm working on the tracker side of code (since some changes will be needed there to support mio). I think it's a good goal to at the very least provide documentation for all structs/fields (since it'll give some pointers about the code). I'm happy to review any PRs related to documentation and try to explain things that are unclear. For the time being I'll leave this bug open as a tracker for improving docs.
